velox/functions/sparksql/String.h

```diff
@@ -293,6 +292,62 @@ struct EndsWithFunction {
   }
 };
 
+/// locate function
+/// locate(string, string) -> integer
+/// locate(string, string, integer) -> integer
+///
+/// Returns the position of the first occurrence of the first string in the
+/// second string after the give position.
+template <typename T>
+struct LocateFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE bool call(
+      out_type<int32_t>& result,
+      const arg_type<Varchar>& subString,
+      const arg_type<Varchar>& string) {
+    result = stringImpl::stringPosition<false /*isAscii*/>(
+        string, subString, 1 /*instance*/, 1 /*start*/);
+    return true;
+  }
+
+  FOLLY_ALWAYS_INLINE bool callAscii(
```

Comment:
Do we need to define the fucntion with two parameters, I think the related case class in spark is defined with three expressions. So velox should only see the call with 3 parameters.

Reply:
Makes sense. Removed the support for 2 parameters.